### PR TITLE
Canvas medical tests

### PR DESCRIPTION
**Pull request by kubiak85**
PR for https://github.com/awell-health/awell-extensions/issues/209
This PR introduces significant improvements:
- Comprehensive tests.
- Critical Bug Fixes: Addressed crucial logic errors.
- Refined Validation: Improved data validation for enhanced accuracy and reduced errors.

**Important!**  
Before merging this PR, please include an additional change to ensure a smooth integration: https://github.com/awell-health/extensions-core/pull/22

### CODE REVIEW

File: extensions/canvasMedical/validation/dto/reference.zod.ts
Comment (bejoinka) [inline extensions/canvasMedical/validation/dto/reference.zod.ts:1]: can i ask? this reference looks different now, as though it's missing the url at the front. is that right?

File: extensions/canvasMedical/validation/dto/extension.zod.ts
Comment (bejoinka) [inline extensions/canvasMedical/validation/dto/extension.zod.ts:1]: I like how you broke this out into individual extensions. i thought recursion was possible?

**Review by bejoinka — COMMENTED** [review]
Looks good. I'm interested to learn more about how you went about finding those schemas. Thanks for your review and work there.

**Comment by danijel** [comment]
I've merged this in the canvas-medical branch, I want to try and have createPatient working today e2e
@kubiak85 you can still answer JB questions here, but let's do any changes if needed in canvas-medical branch

**Pull request merged**